jwst/cube_build/blot_cube_build.py

```diff
@@ -214,15 +214,15 @@ def blot_images(self):
 
             if self.instrument == 'MIRI':
                 valid3 = np.isfinite(lam_det)
-                good_data = valid3 & pixel_mask
+                good_data1 = valid3 & pixel_mask
+                good_data = np.where(good_data1)
```

Comment:
Instead of computing `valid` for every slice and trying to replace NaNs for every slice, just go ahead and dump everything from `ra` into `ra_det` (and the others), and then do a single call at the end, replacing NaNs with zeros:
```
notvalid = np.isnan(lam_det)
lam_det[notvalid] = 0
ra_det[notvalid] = 0
dec_det[notvalid] = 0
flag_det[lam_det != 0.] = 1
```
or something similar.

Btw, `np.zeros` produces floats by default, so `flag_det` above has a `dtype='float64'` by default.  If you want it to be an int, you need
```
flag_det = np.zeros((2048, 2048), dtype=np.int)
```

Reply:
I would rather not make such a change now with the build 7.2 due - the latest tomorrow. It took me a long time to get this figured out for NIRSPEC because I had to do things slice by slice. If I make a change to it now it takes a while to do all the testing needed. I will put it on the list of first things to do after 7.2.